doc/trace-server/trace-server-dev-guide.md

```diff
@@ -0,0 +1,2916 @@
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.tracecompass.tmf.core.component.DataProviderConstants;
+import org.eclipse.tracecompass.tmf.core.config.ITmfConfiguration;
+import org.eclipse.tracecompass.tmf.core.event.ITmfEvent;
+import org.eclipse.tracecompass.tmf.core.event.ITmfLostEvent;
+import org.eclipse.tracecompass.tmf.core.event.aspect.TmfBaseAspects;
+import org.eclipse.tracecompass.tmf.core.filter.ITmfFilter;
+import org.eclipse.tracecompass.tmf.core.filter.model.TmfFilterMatchesNode;
+import org.eclipse.tracecompass.tmf.core.model.CommonStatusMessage;
+import org.eclipse.tracecompass.tmf.core.model.tree.ITmfTreeDataProvider;
+import org.eclipse.tracecompass.tmf.core.model.tree.TmfTreeDataModel;
+import org.eclipse.tracecompass.tmf.core.model.tree.TmfTreeModel;
+import org.eclipse.tracecompass.tmf.core.request.ITmfEventRequest;
+import org.eclipse.tracecompass.tmf.core.request.TmfEventRequest;
+import org.eclipse.tracecompass.tmf.core.response.ITmfResponse;
+import org.eclipse.tracecompass.tmf.core.response.ITmfResponse.Status;
+import org.eclipse.tracecompass.tmf.core.response.TmfModelResponse;
+import org.eclipse.tracecompass.tmf.core.timestamp.TmfTimeRange;
+import org.eclipse.tracecompass.tmf.core.trace.ITmfTrace;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Simple events statistics data provider
+ * 
+ * @author Bernd Hufmann
+ */
+@SuppressWarnings("null")
+@NonNullByDefault
+public class ExampleConfigurableDataTreeDataProvider implements ITmfTreeDataProvider<TmfTreeDataModel> {
+    private static final String BASE_ID = "org.eclipse.tracecompass.examples.nomodulestats.config"; //$NON-NLS-1$;
+    
+    private static long fCount = 0;
+    
+    private @Nullable ITmfTrace fTrace;
+    private @Nullable StatsPerTypeRequest fRequest;
+    private @Nullable List<TmfTreeDataModel> fCachedResult = null;
+    private @Nullable ITmfConfiguration fConfiguration;
+    private String fId = BASE_ID;
+
+    /**
+     * Constructor
+     * @param trace
+     *          the trace (not experiment)
+     */
+    public ExampleConfigurableDataTreeDataProvider(ITmfTrace trace) {
+        this(trace, null);
+    }
+
+    /**
+     * Constructor with configuration
+     * @param trace
+     *      the trace (not experiment)
+     * @param configuration
+     *      the configuration
+     */
+    public ExampleConfigurableDataTreeDataProvider(ITmfTrace trace, @Nullable ITmfConfiguration configuration) {
+        fTrace = trace;
+        fConfiguration = configuration;
+        if (configuration != null) {
+            fId = BASE_ID + DataProviderConstants.ID_SEPARATOR + configuration.getId();
```

Comment:
CONFIG_SEPARATOR

Reply:
In this example it has to be the ID_SEPARATOR because it needs to be the secondary ID of the data provider ID that the DataProviderManager is extracting by API contract.